File: hw/palomo/u3/hw17OOP/cell_phone_project/ec_edu_espe_model/device.py
from abc import ABC, abstractmethod

class Device(ABC):

    def __init__(self, brand, model):
        self._brand = brand
        self._model = model

    @abstractmethod
    def turn_on(self):
        pass

    @abstractmethod
    def turn_off(self):
        pass

    @property
    def brand(self):
        return self._brand

    @property
    def model(self):
        return self._model

File: hw/palomo/u3/hw17OOP/Phone.TypeScript/ec.edu.espe.model/Device.ts
export abstract class Device {
    constructor(
        private brand: string,
        private model: string
    ) {}

    public getBrand(): string {
        return this.brand;
    }

    public getModel(): string {
        return this.model;
    }

    abstract turnOn(): void;
    abstract turnOff(): void;
}